core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -346,15 +346,16 @@ private[spark] class TaskSchedulerImpl(
 
   def statusUpdate(tid: Long, state: TaskState, serializedData: ByteBuffer) {
     var failedExecutor: Option[String] = None
+    var reason: ExecutorLossReason = null
```

Comment:
Nit: it would be clearer to make this an Option similar to `failedExecutor` rather than a null. The two variables are used together.